test/__tests__/hello-world.test.tsx

```diff
@@ -0,0 +1,24 @@
+import '@testing-library/jest-dom'
+import { render } from '@testing-library/react'
+import HelloWorldApp from '@/app/hello-world-app/page'
+import HelloWorldPage from '@/pages/hello-world-page'
+
+describe('Hello World test', () => {
+  it('renders a heading containing the right text', () => {
+    // No good way to render layout here without manually rendering,
+    // and updating the container option to not be a `body` tag
+    const appScreen = render(<HelloWorldApp />)
+    const appHeading = appScreen.getByRole('heading', { level: 1 })
+    // Not working as expected with tailwind styles. Styles are default.
+    // const styles = window.getComputedStyles()
+
+    // Kinda weird I have to do this.
+    appScreen.unmount()
```

Comment:
Testing library `render` calls reuse the same context, and render the component in isolation. The comments in this file are my main hesitations with using Jest.

Reply:
I think that the reason things aren't working as you would want them to is that we're using the wrong tool for the job here. `@testing-library/react` is designed to test React components in isolation but based on the comments it seems you were expecting things _outside_ of the React component, like Tailwind CSS files, to be considered.

I think that we need to define the intent of what we're trying to do. If we're trying to write tests from the perspective of the end-user, visiting a browser, then Playwright can give us that. But if we're looking to write unit tests where we can see how different props result different component behavior, then `@testing-library/react` is very good for that.

I do agree it's weird we have to do `appScree.unmount()`. It makes sense since `render` seems to be using shared global state but it would be nice if there was a way to set up a new `appScreen` on each test and have that be isolated. Perhaps that's an option?

Reply:
Talked about in person. Think this is resolved